lib/src/equatable.dart

```diff
@@ -18,7 +18,7 @@ import './equatable_utils.dart';
 /// ```
 /// {@endtemplate}
 @immutable
-abstract class Equatable {
+abstract mixin class Equatable {
```

Comment:
As per my issue, I still doubt that this should be a class at all. This works perfectly fine as a mixin-only, so I see no reason as to why would anyone extend it instead of mixin it in.

Reply:
It would be quite inconvenient for existing apps if we forced them to refactor all classes to use equatable as a mixin imo

Reply:
fair enough, for sake of compatibility it makes sense. In the long run though (and since it is a breaking change anyways) I don't see value in it being a class.

Reply:
I didn't explain myself well enough: an abstract mixin class can be `extends` equally well as an abstract class. 
Users can update it without change their code.

<img width="923" alt="image" src="https://github.com/felangel/equatable/assets/59682979/affc4609-589f-4042-acac-215b3b6169fd">


https://dart.dev/language/modifier-reference

Reply:
up on that

Reply:
@felangel even in a large project, this change should be easy to make, a simple search and replace can do 95% of the work, plus people can choose to update the version or not.

Reply:
This is still a breaking change because even as a `mixin class` you won't be able to `extend Equatable` in cases like this:
<img width="813" alt="Screenshot 2024-05-26 at 1 45 23 PM" src="https://github.com/felangel/equatable/assets/8855632/4720cd81-fd8e-4163-9afe-a9bac325c245">